web-ui/src/main/resources/catalog/components/common/needhelp/NeedHelpDirective.js

```diff
@@ -152,20 +152,30 @@
             if (gnGlobalSettings.lang !== "en") {
               baseUrl = scope.helpBaseUrl.replace("{{lang}}", gnGlobalSettings.lang);
             } else {
-              baseUrl = scope.helpBaseUrl.replace("/{{lang}}", "");
+              baseUrl = scope.helpBaseUrl.replace("{{lang}}", "");
```

Comment:
I think this should be "."

i.e. there are 2 urls.
- If `lang !== "en"` i.e. fr
   - https://docs.geonetwork-opensource.org/4.2/{{lang}}/api/search/
   will be changed to 
   - https://docs.geonetwork-opensource.org/4.2/fr/api/search/
- else   
   - https://docs.geonetwork-opensource.org/4.2/{{lang}}/api/search/
   will be changed to 
   - https://docs.geonetwork-opensource.org/4.2//api/search/
   which does not look as pretty but still works. The double // in the url remains but if you put it as a "." then you get
   - https://docs.geonetwork-opensource.org/4.2/./api/search/
   and the browser will replace this with 
   - https://docs.geonetwork-opensource.org/4.2/api/search/

Reply:
Now that I think about it, it does not make sense to user `"."` either because someone could use `?lang={{lang}}` and it would be odd to add "." and `?lang=.` would not make sense!
So I guess "" is better in this case.
Maybe there could be another replace on the url after the replace to remove all the double `"//"`

As for why there is a odd case where lang = ""
What if it treated `gnGlobalSettings.lang !== "en"` as if `gnGlobalSettings.lang = null`?
Then `{{lang?en}}` could mean that if lang is null then default to `en`
So you could add another replacement for the following.
`baseUrl = scope.helpBaseUrl.replace("{{lang?en}}", "");`


So gn could use 
`https://docs.geonetwork-opensource.org/4.2/{{lang}}/api/search/`
And other sites that always want to have a language value for language could use 
`https://example.com/{{lang?en}}/api/search/?lang={{lang?en}}` and it would result to en value.
But this would still be odd logic.